### DISCUSSION TITLE
List of contributors to a file truncates with file renames.

### DISCUSSION THREAD

**Discussion by jenstroeger** [community/community#40744] [Repositories]
### Select Topic Area

Bug

### Body

Recently we renamed a file in a public repository (commit [770a3d9](https://github.com/jenstroeger/python-package-template/commit/770a3d969596b73b9ad4a2e040b064b2e09909aa)) and as a result the list of contributors to that file was truncated, and so was its [history](https://github.com/jenstroeger/python-package-template/commits/main/.github/workflows/_build.yaml) (which now sports the “Browse history” link). Looks related to discussion https://github.com/community/community/discussions/3272, which supposedly was resolved earlier this year?

Locally I can reproduce the behavior:
```
git log -- _build.yaml
```
shows two commits to _that_ file, whereas
```
git log --follow -- _build.yaml
```
[follows](https://www.git-scm.com/docs/git-log#Documentation/git-log.txt---follow) the history of the file beyond renames and therefore covers _all_ contributors (and history).

Which raises two questions:
- What’s the definition of a “contributor” to a file on Github?
- How is the list of contributors to a single file (and the file’s history) implemented by GitHub?

I’d argue that the list of contributors should always `--follow`.

**Comment by ossanna16** [comment]
Hi there @jenstroeger and welcome to our community! Thank you for asking a great question 🙂
While you are waiting for a reply to your question, here are a few things you can do:

To get started, introduce yourself in our [official introduction thread](https://github.com/orgs/community/discussions/23204)
To connect with other new users, check out this Discussion to [Meet Your GitHub Community Team](https://github.com/orgs/community/discussions/28776), further your GitHub knowledge by working through our [GitHub Skill 🆙 exercises](https://github.com/orgs/community/discussions/32121), and participate in our [weekly Fun Friday chats](https://github.com/orgs/community/discussions/30912)

**Comment by Bullin112** [comment]
It appears that the bug you have encountered is related to the way GitHub handles file renaming. When a file is renamed, GitHub may not properly track the full history of the file and may truncate the list of contributors. This can lead to the behavior you described where the history of the file is not fully displayed.

I would recommend reporting this bug to GitHub support so that they can investigate and fix the issue. In the meantime, using the --follow option with git log should allow you to see the full history and list of contributors for the file.

For other info check out this [page](https://tennisdig.com/tennis-ball-machine-diy/).
